src/main/kotlin/uk/gov/dluhc/printapi/service/CertificateBatchingService.kt

```diff
@@ -29,12 +36,41 @@ class CertificateBatchingService(
 
     fun batchCertificates(batchSize: Int): Map<String, List<Certificate>> {
         val certificatesPendingAssignment = certificateRepository.findByStatus(Status.PENDING_ASSIGNMENT_TO_BATCH)
-        return certificatesPendingAssignment.chunked(batchSize).associate { batchOfCertificates ->
+
+        return limitCertificates(certificatesPendingAssignment).chunked(batchSize).associate { batchOfCertificates ->
             val batchId = idFactory.batchId()
             batchId to batchOfCertificates.onEach {
                 it.getCurrentPrintRequest().batchId = batchId
                 it.addStatus(Status.ASSIGNED_TO_BATCH)
             }
         }
     }
+
+    private fun limitCertificates(certificatesPendingAssignment: List<Certificate>): List<Certificate> {
+        val currentDate = LocalDate.ofInstant(Instant.now(clock), ZoneOffset.UTC)
+        val startOfDay = currentDate.atStartOfDay().toInstant(ZoneOffset.UTC)
+        val endOfDay = currentDate.plusDays(1).atStartOfDay().toInstant(ZoneOffset.UTC).minusSeconds(1)
+        val countOfRequestsSentToPrintProvider =
+            certificateRepository.getPrintRequestStatusCount(startOfDay, endOfDay, Status.ASSIGNED_TO_BATCH)
+
+        if (certificatesPendingAssignment.size + countOfRequestsSentToPrintProvider> dailyLimit) {
+            logDailyLimit(certificatesPendingAssignment, countOfRequestsSentToPrintProvider)
+            return certificatesPendingAssignment.subList(0, dailyLimit - countOfRequestsSentToPrintProvider)
```

Comment:
Since it is easier to calculate how many certificates can be sent to print provider, I returned the sub-list accordingly. 

It would be more complicated to return the number corresponding only full-sized batches and the log will be more complicated to understand as well.